### PR TITLE
Exclude non-salable subproducts from price calculation

### LINKED ISSUE #1074: Wrong price range for configurable with out of stock subs
Fix configurable products' prices #978 includes sub products that are not salable (out of stock for example) in price range calculation.
This by itself is possibly a desired behavior but a further bug happens here when the subproducts are setup with special prices... for example we have a configurable with 2 sub products.
All are priced $100, with special price without end date $50.
When all the products are in stock the price calculation gives the $50 value. (which is correct)
When one of the subs goes out of stock the getFinalPrice call on him returns the full price value, and in the end the price calculation gives the "$50 - $100" value (which is incorrect).

I've opened a PR with a suggested fix.

### PR DESCRIPTION
**Pull request by Interghost**
Fixes #1074

### CODE REVIEW

File: app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php [hunk 1]
```diff
@@ -600,15 +600,19 @@ protected function handlePrice(Mage_Catalog_Model_Product &$product, $sub_produc
                     } else {
                         if (count($sub_products) > 0) {
                             foreach ($sub_products as $sub_product) {
+                                if (!$sub_product->isSalable()) {
+                                    continue;
+                                }
                                 $price = (double) $taxHelper->getPrice($product, $sub_product->getFinalPrice(), $with_tax,
```
Comment (JanPetr) [inline app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php]: Could you add an empty line before `$price = ..." line?

File: app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php [hunk 2]
```diff
@@ -600,15 +600,19 @@ protected function handlePrice(Mage_Catalog_Model_Product &$product, $sub_produc
...
                                                          null, null, null, $product->getStore(), null);
 
                                 $min = min($min, $price);
                                 $max = max($max, $price);
                             }
-                        } else {
+                        }
+                        if ($min > $max) {
```
Comment (JanPetr) [inline app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php]: Could you add an empty line before `if (..." line?

**Comment by Interghost** [comment]
Done

**Comment by JanPetr** [comment]
You are awesome, thanks! 🚀

**Pull request merged**